### PR TITLE
Add -JsonFile parameter to Test-Json command issue #12882

### PR DESCRIPTION
**Pull request by Liturgist**
# PR Summary
Add -JsonFile parameter to Test-Json command issue #12882
## PR Context
JSON data sets are often stored as files. The recent addition of -SchemaFile makes -JsonFile desireable.
## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
@Liturgist Please fix CodeFactor issues.

**Comment by Liturgist** [comment]
> @Liturgist Please fix CodeFactor issues.

To avoid the CodeFactor complaint, I put a blank line before the comment. Is there a better way?
```
            catch (Exception e) when (

                // Handle exceptions related to file access to provide more specific error message
                // https://docs.microsoft.com/en-us/dotnet/standard/io/handling-io-errors
                e is IOException ||
                e is UnauthorizedAccessException ||

```

**Comment by Liturgist** [comment]
> @Liturgist Please fix CodeFactor issues.

It appears that I have satiated CodeFactors needs. But, there is still one test running for ~10 hours. Is there anything I need to do?

**Comment by iSazonov** [comment]
I restarted CI-MacOS.

**Comment by Liturgist** [comment]
> I restarted CI-MacOS.

Thank you for your help. Is this something I could have done? If so, where can I read about how to do it?

**Comment by iSazonov** [comment]
@Liturgist I used my maintainer permissions.

**Comment by Liturgist** [comment]
> @Liturgist I used my maintainer permissions.

When might this PR be merged? I realize this is not a critical change.

**Comment by Liturgist** [comment]
I had to change global.json to use the version of .Net that I have on my machine. I did not think I was checking that back in or making it part of the PR.

1) How do I get my global.json out of this PR?
2) How do I get the .Net that PowerShell/PowerShell is using?

**Comment by iSazonov** [comment]
@Liturgist You should rebase.

**Comment by Liturgist** [comment]
@iSazonov, I have read a few things about rebase, but it remains a bit unclear to me. Can you suggest any specific documentation?

**Comment by iSazonov** [comment]
@Liturgist 
1. Checkout the work branch
2. git pull --rebase PowerShell master
3. Resolve merge conflicts
4. Follow git help message and continue rebasing

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs [hunk 1]
```diff
@@ -78,6 +92,44 @@ protected override void BeginProcessing()
...
+                if (Json != null)
+                {
+                    _json = Json;
+                }
+                else if (JsonFile != null)
+                {
+                    try
+                    {
+                        resolvedpath = Context.SessionState.Path.GetUnresolvedProviderPathFromPSPath(JsonFile);
+                        // _json = JsonFile.FromFileAsync(resolvedpath).Result;
```
Comment (adityapatwardhan) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs:106]: Please remove commented line.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Test-Json.Tests.ps1 [hunk 2]
```diff
@@ -65,48 +64,100 @@ Describe "Test-Json" -Tags "CI" {
...
+        Set-Content -Path $validSchemaJsonPath -Value $validSchemaJson -Encoding utf8BOM
+        Set-Content -Path $invalidSchemaJsonPath -Value $invalidSchemaJson -Encoding utf8BOM
+    }
 
     It "Missing JSON schema file doesn't exist" {
         Test-Path -LiteralPath $missingSchemaJsonPath | Should -BeFalse
     }
 
+    It "Missing JSON file doesn't exist" {
+        Test-Path -LiteralPath $missingJsonPath | Should -BeFalse
```
Comment (adityapatwardhan) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Test-Json.Tests.ps1:89]: Should this be `Test-Json` ?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**